code/modules/projectiles/gun.dm

```diff
@@ -210,6 +210,10 @@
 	fire_sounds()
 	if(suppressed || !message)
 		return FALSE
+
+	for(var/obj/item/taperecorder/rec in get_hearers_in_view(DEFAULT_MESSAGE_RANGE, src))
+		rec.hear_raw("making gunshot!", src)
```

Comment:
With the proposed refactor above we could easily make a proc for sound events
```dm
/proc/hearable_sound_event(message, atom/source, range)
	for(var/atom/hearer as anything in get_hearers_in_view(range, source))
		// We'd need to add some flag that blocks printing to chat unless you're blind
		// (Otherwise we'd spam people's chats)
		hearer.Hear(message, source) 
```
```dm
hearable_sound_event("You hear a gunshot!", src, COMBAT_MESSAGE_RANGE)
```
This would handle everything, making it suuuper easy to add "sounds" that things like the tape recorder can react to.
It'd also let us remove a lot of `blind_message`s from `visible_messages` which is a bit cleaner.